src/ray/object_manager/object_buffer_pool.cc

```diff
@@ -179,6 +170,66 @@ std::vector<ObjectBufferPool::ChunkInfo> ObjectBufferPool::BuildChunks(
   return chunks;
 }
 
+ray::Status ObjectBufferPool::EnsureBufferExists(
+    const ObjectID &object_id, const rpc::Address &owner_address, uint64_t data_size,
+    uint64_t metadata_size, uint64_t chunk_index, std::unique_lock<std::mutex> &lock) {
+  RAY_CHECK(lock.owns_lock());
+
+  // Buffer for object_id already exists.
+  if (create_buffer_state_.contains(object_id)) return ray::Status::OK();
```

Comment:
I think our style convention is to always put returns on a new line with braces, and never inline.